data/json/recipes/food/dairy_products.json

```diff
@@ -37,5 +37,23 @@
     "autolearn": true,
     "qualities": [ { "id": "COOK", "level": 1 } ],
     "components": [ [ [ "milk_cream_jar", 1 ] ] ]
+  },
+  {
+    "type": "recipe",
+    "result": "ghee",
+    "category": "CC_FOOD",
+    "subcategory": "CSC_FOOD_DAIRY",
+    "skill_used": "cooking",
+    "difficulty": 2,
+    "charges": 1,
+    "time": "20 m",
+    "autolearn": true,
+    "batch_time_factors": [ 80, 4 ],
+    "qualities": [ { "id": "BOIL", "level": 1 }, { "id": "COOK", "level": 2 } ],
+    "tools": [
+      [ [ "water_boiling_heat", 2, "LIST" ] ],
+      [ [ "colander_steel", -1 ], [ "sieve_steel", -1 ], [ "sieve_primitive", -1 ], [ "rag", -1 ] ]
```

Comment:
It doesn't strike me that a rag work work like any of the rest of these.

Reply:
Well, we don’t have cheesecloth in the game, and some fabrics could stand in for it, I’m sure. Rags are gauze-like to make bandages from, so I figure it could work

Reply:
I think it'd be best to assume normal rags don't work, and allow only gauze-like things (and/or add  cheesecloth).
I don't think so very strongly, so I don't really care too which way you decide to go.

Reply:
How do you think cheesecloth ought to work? If you can craft rags from thread without a loom then I’d assume that you could either start with thread and just not finish a rag, or disassemble a rag, pulling out some but not all its threads. Should be disassembled into thread I suppose. Where else could we use cheesecloth? Cheese, obviously, but does anything else come to mind?

Reply:
Looking at hard cheese's recipe, it uses a rag as a tool for the same thing. I could add cheesecloth, but I want to keep this PR as small as possible. I'll think about putting it in another PR

Reply:
I don't know how you make rags in real life, but I'd suggest tracking a real recipe with the cheesecloth recipe. Nothing else particularly comes to mind.

Since there's precedent for it, yeah, just follow that.